code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/CarouselTemplateUtil.kt

```diff
@@ -0,0 +1,115 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification
+
+import android.content.Intent
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ui.notification.models.CarouselPushTemplate
+import org.json.JSONArray
+import org.json.JSONException
+
+internal object CarouselTemplateUtil {
+    private const val SELF_TAG = "CarouselTemplateHelpers"
+
+    /**
+     * Calculates a new left, center, and right index given the current center index, total number
+     * of images, and the intent action.
+     *
+     * @param centerIndex [Int] containing the current center image index
+     * @param listSize `Int` containing the total number of images
+     * @param action [String] containing the action found in the broadcast [Intent]
+     * @return [List] containing the new calculated left, center, and right indices
+     */
+    internal fun calculateNewIndices(
```

Comment:
Please see my comment on consolidating `FilmstripCarouselNotificationBuilder` and `ManualCarouselNotificationBuilder` into a single builder `ManualCarouselNotificationBuilder`. If that is done, we can move this method there.

Reply:
I am wondering if we can split this into two methods `calculateNextIndices` and `calculatePreviousIndices`. Since we are concerned with only 3 images, we can also use the `Triple` structure in Kotlin.